src/sparse_blas/backends/cusparse/CMakeLists.txt

```diff
@@ -0,0 +1,85 @@
+#===============================================================================
+# Copyright 2024 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+#
+# SPDX-License-Identifier: Apache-2.0
+#===============================================================================
+
+set(LIB_NAME onemkl_sparse_blas_cusparse)
+set(LIB_OBJ ${LIB_NAME}_obj)
+
+include(WarningsUtils)
+
+add_library(${LIB_NAME})
+add_library(${LIB_OBJ} OBJECT
+  cusparse_handles.cpp
+  cusparse_scope_handle.cpp
+  operations/cusparse_spmm.cpp
+  operations/cusparse_spmv.cpp
+  operations/cusparse_spsv.cpp
+  $<$<BOOL:${BUILD_SHARED_LIBS}>: cusparse_wrappers.cpp>
+)
+add_dependencies(onemkl_backend_libs_sparse_blas ${LIB_NAME})
+
+target_include_directories(${LIB_OBJ}
+  PRIVATE ${PROJECT_SOURCE_DIR}/include
+          ${PROJECT_SOURCE_DIR}/src
+          ${CMAKE_BINARY_DIR}/bin
+          ${ONEMKL_GENERATED_INCLUDE_PATH}
+)
+
+target_compile_options(${LIB_OBJ} PRIVATE ${ONEMKL_BUILD_COPT})
+
+if (${CMAKE_VERSION} VERSION_LESS "3.17.0")
+  find_package(CUDA 12.2 REQUIRED)
```

Comment:
You might want to upgrade this 12.4.0 as a minimum. Reason being CUDA 12.4 introduced the [`cusparseSpMV_preprocess`](https://docs.nvidia.com/cuda/archive/12.4.0/cusparse/index.html#cusparsespmv) that would be part of oneMKL Interfaces implementation that previous versions, e.g., [CUDA 12.3.2](https://docs.nvidia.com/cuda/archive/12.3.2/cusparse/index.html#cusparsespmv), did not have.
If we must support a slightly older version of CUDA, then in `spmv_optimize` call, we will need to add some `#ifdef` check on CUDA version to decide whether to call `cusparseSpMV_preprocess` or not.